tools/ui-components/src/button/types.ts

```diff
@@ -4,12 +4,12 @@ export type ButtonVariant = 'primary' | 'danger' | 'info';
 
 export type ButtonSize = 'small' | 'medium' | 'large';
 
-export interface ButtonProps
-  extends React.ButtonHTMLAttributes<HTMLButtonElement | HTMLAnchorElement> {
+export interface ButtonProps<TElement = HTMLButtonElement>
+  extends React.ButtonHTMLAttributes<TElement> {
   children: React.ReactNode;
   variant?: ButtonVariant;
   size?: ButtonSize;
-  onClick?: MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+  onClick?: MouseEventHandler<TElement>;
```

Comment:
has to change the type, to fix this error

![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/88248797/38903d1a-0092-4a56-9655-20277dd4049e)